### PR TITLE
fix(api): persist tool call counts across V1 runs

### PR DESCRIPTION
**Pull request by alecf**
## Summary

- V1 API passed `{}` as `toolCallCounts` to `advanceThread` on every run, so counts reset across client-side tool execution rounds. This broke both per-tool `maxCalls` limits and the project-wide `maxToolCallLimit`.
- Added `buildToolCallCountsFromMessages()` that reconstructs tool call counts from thread message history, scoped to the current round (since last user message).
- V1 `executeRun` now fetches thread messages and passes reconstructed counts to `advanceThread` instead of `{}`.

Fixes TAM-1151

## Test plan

- [x] Unit tests for `buildToolCallCountsFromMessages` covering empty messages, no tool calls, single/multiple tool calls, round scoping, parameter differentiation, and edge cases (7 new tests, all passing)
- [ ] Verify per-tool `maxCalls` limits are enforced across V1 runs (client executes tool, sends results back, limit still tracked)
- [ ] Verify project-wide `maxToolCallLimit` is enforced across V1 runs

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: apps/api/src/v1/v1.service.ts [hunk 1]
```diff
@@ -735,6 +736,13 @@ export class V1Service {
 
           const forceToolChoice = convertV1ToolChoiceToInternal(dto.toolChoice);
 
+          // Reconstruct tool call counts from thread history so limits persist
+          // across V1 runs (client executes a tool, sends results back).
+          const dbMessages = await operations.getMessages(this.db, threadId);
+          const threadMessages = dbMessages.map(dbMessageToThreadMessage);
+          const toolCallCounts =
+            buildToolCallCountsFromMessages(threadMessages);
+
```
[the remark this answers is not in the document]
Comment (alecf) [reply apps/api/src/v1/v1.service.ts:745]: this is fine for now

[the remark this answers is not in the document]
Comment (alecf) [reply apps/api/src/v1/v1.service.ts:745]: @CharlieHelps operations.getMessages() already has orderBy - does that fix this?
